### PR TITLE
introduce namespace migration and do fixes

### PR DESCRIPTION
**Pull request by The-Voidwalker**
This should introduce:
* Namespace deletion
* Namespace renaming (because this didn't work before)
* Namespace migration (handles conflicts in page titles and moving entire namespaces around)

This fixes:
* Default edit restriction is set to the permission "none" ([Example](https://publictestwiki.com/wiki/Special:ListGroupRights#Namespace_restrictions))
* The error produced by [T3963](https://phabricator.miraheze.org/T3963)
* An issue created by $htmlForm being `false` in SpecialManageWikiNamespaces.php
* Any page becoming inaccessible when a namespace is added or modified

### CODE REVIEW

**Comment by JohnFLewis** [comment]
In general okay besides missing i18n and the changes to $existingNamespace - you want to query by name not ID.

File: includes/formFactory/ManageWikiFormFactory.php [hunk 1]
```diff
@@ -380,60 +412,29 @@ protected function submitForm(
...
-					$nsAlias = (string)json_encode( explode( "\n", $formData["aliases-$name"] ) );
-				}
 				$build[$name] = [
 					'ns_dbname' => $wgDBname,
 					'ns_namespace_id' => $nsID[$name],
 					'ns_namespace_name' => str_replace( ' ', '_', $formData["namespace-$name"] ),
 					'ns_searchable' => (int)$formData["search-$name"],
 					'ns_subpages' => (int)$formData["subpages-$name"],
-					'ns_aliases' => $nsAlias,
+					'ns_aliases' => $formData["aliases-$name"] == "" ? "[]" : json_encode( explode( "\n", $formData["aliases-$name"] ) ),
```
Comment (paladox) [inline includes/formFactory/ManageWikiFormFactory.php]: Try empty() ? <> : <>;

**Comment by paladox** [comment]
https://gerrithub.io/c/miraheze/ManageWiki/+/10

https://gerrithub.io/c/miraheze/ManageWiki/+/11

https://gerrithub.io/c/miraheze/ManageWiki/+/12

I've imported the pull into GerritHub

File: includes/formFactory/ManageWikiFormFactory.php (hunk 1 above)
Comment (The-Voidwalker) [reply includes/formFactory/ManageWikiFormFactory.php]: To what end? This does the same thing, and I know it works.

**Comment by paladox** [comment]
merged on gerrithub.

**Pull request closed without merging**